test/typescript/styles.spec.tsx

```diff
@@ -109,18 +109,18 @@ const AllTheComposition = withTheme(
 // Can't use withStyles effectively as a decorator in TypeScript
 // due to https://github.com/Microsoft/TypeScript/issues/4881
 // @withStyles(styles)
-class DecoratableComponent extends React.Component<WithStyles<StyledComponentProps, 'root'>> {
-  render() {
-    const { classes, text } = this.props;
-    return (
-      <div className={classes.root}>
-        {text}
-      </div>
-    );
+const DecoratedComponent = withStyles(styles)(
+  class extends React.Component<WithStyles<StyledComponentProps, 'root'>> {
+    render() {
+      const { classes, text } = this.props;
+      return (
+        <div className={classes.root}>
+          {text}
+        </div>
+      );
+    }
   }
-}
-
-const DecoratedComponent = withStyles(styles)(DecoratableComponent);
+);
```

Comment:
This is currently the least painful way in TypeScript that you can use a type-mutating decorator on a class.